server/service/jct/api.coffee

```diff
@@ -1417,3 +1426,130 @@ API.service.jct.test = (params={}) ->
   return res
 
 
+# return the institution for an id. Import the data if refresh is true
+API.service.jct.institution = (id, refresh) ->
+  if refresh
+    console.log('Got refresh - importing institution (ror)')
+    Meteor.setTimeout (() => API.service.jct.institution.import()), 1
+    return true
+  if id
+    rec = jct_institution.find 'id.exact:"' + id.toLowerCase().trim() + '"'
+    if rec
+      return rec
+    return {}
+  else
+    return total: jct_institution.count()
+
+# import institution data from ror
+API.service.jct.institution.import = () ->
+  _set_id = (rec) ->
+    ror = rec.id
+    id = ror.replace("https://ror.org/", '')
+    rec.id = id
+    rec.ror = id
+    rec.ror_id = ror
+    return rec
+
+  _set_title = (rec) ->
+    rec.title = rec.name
+    return rec
+
+  # Download zip file from github containing data dump of RoR
+  fldr = '/tmp/jct_ror' + (if API.settings.dev then '_dev' else '') + '/'
+  if fs.existsSync fldr
+    fs.rmdirSync(fldr, { recursive: true });
+  fs.mkdirSync fldr
+  filepath = fldr + 'ror.zip'
+  ror_data_dump_url = 'https://github.com/ror-community/ror-api/raw/master/rorapi/data/ror-2021-09-23/ror.zip'
+  fs.writeFileSync filepath, HTTP.call('GET', ror_data_dump_url, {npmRequestOptions:{encoding:null}}).content
+  console.log 'Importing from ROR data dump ' + fldr + 'ror.zip'
+  zip = AdmZip(filepath)
+  zip.extractEntryTo("ror.json", fldr)
+
+  # Remove old data and index new
+  removed = false
+  batch = []
+  counter = 1
+  imports = 0
+  for rec in JSON.parse fs.readFileSync fldr + 'ror.json'
+    if batch.length >= 20000
+      if not removed
+        console.log 'Removing old institution records'
+        jct_institution.remove '*'
+        future = new Future()
+        Meteor.setTimeout (() -> future.return()), 10000
+        future.wait()
+        removed = true
+      console.log 'Importing RoR batch ' + counter
+      jct_institution.insert batch
+      counter += 1
+      batch = []
+    rec = _set_id(rec)
+    rec = _set_title(rec)
+    batch.push rec
+    imports += 1
+  if batch.length
+    console.log 'Importing RoR batch ' + counter
+    jct_institution.insert batch
+    batch = []
+  console.log('Completed importing ror data ', imports)
+
+_ror_api_import = () ->
```

Comment:
This method doesn't seem to be used anywhere

Reply:
Yes, beacuse I started using the api, but realised it will only do 10,000 records. Keeping it in code so we have it if we want to use the api later